test/e2e/specs/site-editor/preload.spec.js

```diff
@@ -16,26 +16,38 @@ test.describe( 'Preload', () => {
 		page,
 		admin,
 	} ) => {
-		// Do not use `visitSiteEditor` because it waits for the iframe to load.
-		await admin.visitAdminPage( 'site-editor.php' );
-
 		const requests = [];
-		let isLoaded = false;
 
-		page.on( 'request', ( request ) => {
-			if ( request.resourceType() === 'document' ) {
-				// The iframe also "requests" a blob document. This is the most
-				// reliable way to wait for the iframe to start loading.
-				// `waitForSelector` is always a bit delayed, and we don't want
-				// to detect requests after the iframe mounts.
-				isLoaded = true;
-			} else if ( ! isLoaded && request.resourceType() === 'fetch' ) {
-				requests.push( request.url() );
+		function onRequest( request ) {
+			if (
+				request.resourceType() === 'document' &&
+				request.url().startsWith( 'blob:' )
+			) {
+				// Stop recording when the iframe is initialized.
+				page.off( 'request', onRequest );
+			} else if ( request.resourceType() === 'fetch' ) {
+				const url = request.url();
+				const urlObject = new URL( url );
+				const restRoute = urlObject.searchParams.get( 'rest_route' );
+				if ( restRoute ) {
+					requests.push( restRoute );
+				} else {
+					requests.push( url );
+				}
 			}
-		} );
+		}
+
+		page.on( 'request', onRequest );
 
-		await page.waitForFunction( ( _isLoaded ) => _isLoaded, [ isLoaded ] );
+		await admin.visitSiteEditor();
 
-		expect( requests ).toEqual( [] );
+		// To do: these should all be removed or preloaded.
+		expect( requests ).toEqual( [
+			// There are two separate settings OPTIONS requests. We should fix
+			// so the one for canUser and getEntityRecord are reused.
+			'/wp/v2/settings',
```

Comment:
Maybe we should delay the destruction of the preloaded cache by a couple of MS instead of removing it after the first match. Technically, this should help resolve anything in the critical path via cache without "affecting" the `PreloadingMiddleware` behavior.

https://github.com/WordPress/gutenberg/blob/5c76815eb6e47f8cdd8f32c919f431467b078366/packages/api-fetch/src/middlewares/preloading.js#L53-L54

Reply:
Maybe next idle callback? Still, it doesn't feel like clean solution but it could work for now I guess.